miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultBoxService.java

```diff
@@ -519,4 +519,15 @@ public DeletionStore getDeletionStore() {
     return deletionStore;
   }
 
+  @Override
+  public void throwIfBoxPositionIsFilled(Boxable boxable) throws IOException {
+    if (boxable.getBox() == null || boxable.getBoxPosition() == null) return;
+    Box box = get(boxable.getBox().getId());
+    BoxableView boxableView = box.getBoxable(boxable.getBoxPosition());
+    if (boxableView == null) return;
+    if (!(boxableView.getId().getTargetId() == boxable.getId() && boxableView.getName().equals(boxable.getName()))) {
```

Comment:
Compare the target type instead of name: `boxableView.getId().getTargetType() == boxable.getEntityType()`